compiler/parser/parser.peg

```diff
@@ -1293,6 +1294,15 @@ Tuple
       }, nil
     }
 
+SQLTimestamp
+  = TIMESTAMP _ s:StringLiteral {
+        return &ast.SQLTimestamp {
+          Kind: "SQLTimestamp",
+          Value: s.(*ast.Primitive),
```

Comment:
Some mgbench queries also use SQL DATE syntax. How about handling that, too?

```suggestion
SQLDateTime
  = typ:(DATE / TIMESTAMP) _ value:StringLiteral {
        return &ast.SQLTimestamp {
          Kind: "SQLTimestamp",
          Type: typ,
          Value: value.(*ast.Primitive),
```

Reply:
If we do this, I don't think there's any need to constrain DATE values to exclude a time (since there's no problem for now with accepting the larger language).

Reply:
If we change the name to ast.SQLTimeValue I think we can go this route. Postgres does indeed allow you to parse timestamp values as date however if you have a timestamp string w/ non-zero hh:mm:ss I believe you need to have logic to zero those out if we are going to be consistent w/ date casting in https://github.com/brimdata/super/pull/5695

Reply:
I'm in favor of doing this, with rounding for DATE in semantic.analyzer.semExpr.